### PR TITLE
[JENKINS-67226] Deduplicate RPCRequest oids and proxy objects

### PR DESCRIPTION
**Pull request by daniel-beck**
See [JENKINS-67226](https://issues.jenkins.io/browse/JENKINS-67226).

This cleans up some causes for duplicate entries in the stack traces for agent-to-controller file access via `FilePath` we're collecting telemetry for.

Upstream sources for the `RPCRequest` string we're changing:

* https://github.com/jenkinsci/remoting/blob/91af15b63ef6431621bbd080be183edb8310f141/src/main/java/hudson/remoting/Command.java#L190
* https://github.com/jenkinsci/remoting/blob/92ecf07540c9e2d8ef3c76f8008afcc8a160e505/src/main/java/hudson/remoting/UserRequest.java#L302
* https://github.com/jenkinsci/remoting/blob/91af15b63ef6431621bbd080be183edb8310f141/src/main/java/hudson/remoting/RemoteInvocationHandler.java#L968-L979

The other looks like this in the stack trace:

```
	at com.sun.proxy.$Proxy5.end(Unknown Source)
```

Probably JVM dependent and referencing a proxy instance by ID? This this should at least cut down some of the hits. We can always amend this.

### Proposed changelog entries

* More aggressively deduplicate entries in agent-to-controller file access telemetry introduced in Jenkins 2.322.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
